tests/unit_tests/experiment_plans/test_grid_detection_plan.py

```diff
@@ -265,3 +276,92 @@ def test_when_grid_detection_plan_run_then_grid_detection_callback_gets_correct_
     assert my_grid_params.z_axis == test_z_grid_axis
 
     assert my_grid_params.set_stub_offsets is True
+
+
+@pytest.mark.parametrize(
+    "odd",
+    [(True), (False)],
+)
+@patch("dodal.beamlines.beamline_utils.active_device_is_same_type", lambda a, b: True)
+@patch("bluesky.plan_stubs.sleep", new=MagicMock())
+@patch(
+    "hyperion.experiment_plans.oav_grid_detection_plan.wait_for_tip_to_be_found_ad_mxsc"
+)
+@patch("hyperion.experiment_plans.oav_grid_detection_plan.LOGGER")
+def test_when_detected_grid_has_odd_y_steps_then_add_a_y_step_and_shift_grid(
```

Comment:
I don't know how I feel about this test, it looks horrible to read and is very long

Reply:
I think it's okay. It's a lot of setup but that seems unavoidable for where the stuff is you're testing - in an extremely long function (this is why splitting stuff up into little functions makes it more testable). I don't think it's too bad to read. All of our tests for plans are a bit like this.